modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/frontend/data/set/view/table/ObjectEntriesTableFDSView.java

```diff
@@ -38,94 +51,231 @@
+	}
 
-			if (objectField.getListTypeDefinitionId() > 0) {
-				fieldName = fieldName + ".name";
-			}
+	private void _addFDSTableSchemaField(
+		String contentRenderer, FDSTableSchemaBuilder fdsTableSchemaBuilder,
+		String fieldName, String label, boolean sortable, String type) {
 
-			FDSTableSchemaField fdsTableSchemaField = null;
+		FDSTableSchemaField fdsTableSchemaField = null;
 
-			if (Objects.equals(objectField.getDBType(), "Clob") ||
-				Objects.equals(objectField.getDBType(), "String")) {
+		if (Objects.equals(type, "Clob") || Objects.equals(type, "String")) {
+			StringFDSTableSchemaField stringFDSTableSchemaField =
+				fdsTableSchemaBuilder.addFDSTableSchemaField(
+					StringFDSTableSchemaField.class, fieldName, label);
 
-				StringFDSTableSchemaField stringFDSTableSchemaField =
-					fdsTableSchemaBuilder.addFDSTableSchemaField(
-						StringFDSTableSchemaField.class, fieldName,
-						objectField.getLabel(locale, true));
+			stringFDSTableSchemaField.setTruncate(true);
 
-				stringFDSTableSchemaField.setTruncate(true);
+			fdsTableSchemaField = stringFDSTableSchemaField;
+		}
+		else if (Objects.equals(type, "Date")) {
+			DateFDSTableSchemaField dateFDSTableSchemaField =
+				fdsTableSchemaBuilder.addFDSTableSchemaField(
+					DateFDSTableSchemaField.class, fieldName, label);
 
-				fdsTableSchemaField = stringFDSTableSchemaField;
-			}
-			else if (Objects.equals(objectField.getDBType(), "Date")) {
-				DateFDSTableSchemaField dateFDSTableSchemaField =
-					fdsTableSchemaBuilder.addFDSTableSchemaField(
-						DateFDSTableSchemaField.class, fieldName,
-						objectField.getLabel(locale, true));
+			dateFDSTableSchemaField.setFormat("short");
 
-				dateFDSTableSchemaField.setFormat("short");
+			fdsTableSchemaField = dateFDSTableSchemaField;
+		}
+		else {
+			fdsTableSchemaField = fdsTableSchemaBuilder.addFDSTableSchemaField(
+				fieldName, label);
 
-				fdsTableSchemaField = dateFDSTableSchemaField;
+			if (Objects.equals(type, "Boolean")) {
+				fdsTableSchemaField.setContentRenderer("boolean");
 			}
-			else {
-				fdsTableSchemaField =
-					fdsTableSchemaBuilder.addFDSTableSchemaField(
-						fieldName, objectField.getLabel(locale, true));
+		}
 
-				if (Objects.equals(objectField.getDBType(), "Boolean")) {
-					fdsTableSchemaField.setContentRenderer("boolean");
-				}
-			}
+		if (Validator.isNotNull(contentRenderer)) {
+			fdsTableSchemaField.setContentRenderer(contentRenderer);
+		}
 
-			fdsTableSchemaBuilder.addFDSTableSchemaField(fdsTableSchemaField);
+		if (!Objects.equals(type, "Blob") && sortable) {
+			fdsTableSchemaField.setSortable(true);
+		}
 
-			if (!Objects.equals(objectField.getDBType(), "Blob") &&
-				objectField.isIndexed()) {
+		fdsTableSchemaBuilder.addFDSTableSchemaField(fdsTableSchemaField);
+	}
 
-				fdsTableSchemaField.setSortable(true);
-			}
+	private void _addNonobjectField(
+		FDSTableSchemaBuilder fdsTableSchemaBuilder, String objectFieldName) {
```

Comment:
@rodrigopaulino since it's a non-object field, wouldn't it be better to call `fieldName` instead of `objectFieldName`?

Reply:
I agree! I'll make this change. Thanks, @carolmariaabb